First set of articles

igortg opened the pull request:
List is sorted in order of complexity

Commit:
First set of articles

List is sorted in orderd of complexity

Commit:
Add Empty - The Codeless Code

kfasolin reviewed (approved):
Awesome initiative! 😄

marcelotrevisani commented:
It would be good if we create like a new section for talks and books as well
:)

nicoddemus commented on README.md:
```diff
@@ -1,7 +1,17 @@
 # Awesome Dev Articles [![Awesome](https://cdn.rawgit.com/sindresorhus/awesome/d7305f38d29fed78fa85652e3a63e154dd8e8829/media/badge.svg)](https://github.com/sindresorhus/awesome)
 
-A curated list of articles
+A curated list of articles created by the ESSS Dev team
```
"created"? We didn't write those articles

nicoddemus commented on README.md:
```diff
@@ -1,7 +1,17 @@
+* [Things You Should Never Do, Part I](https://www.joelonsoftware.com/2000/04/06/things-you-should-never-do-part-i/) - Joel Spolsky
+* [The Joel Test: 12 Steps to Better Code](https://www.joelonsoftware.com/2000/08/09/the-joel-test-12-steps-to-better-code/) - Joel Spolsky
+* [Move Fast and Fix Things](https://githubengineering.com/move-fast/) - Vicent Martí
+* [Notes on structured concurrency, or: Go statement considered harmful](https://vorpus.org/blog/notes-on-structured-concurrency-or-go-statement-considered-harmful/) -  Nathaniel J. Smith
```
Please add a section "how to add new articles":

```markdown
### How to add new articles

The idea of this list is to aggregate articles and interesting links that we usually share in the mailing list/RocketChat for easier reference in the future, so new articles are welcome!

To add a new article just open a new PR against the repository, adding the link in the correct position to keep the list sorted (you can open the PR directly in GitHub's UI).
```

nicoddemus commented on README.md:
```diff
@@ -1,7 +1,17 @@
 
 ## Articles
+
+* [Steps to better code](https://medium.com/@isaaclyman/steps-to-better-code-e6c3cce0c7f9) - Isaac Lyman
```
How about sorting this by title or date?

igortg replied:
It's sorted by complexity. My intention was to make it easier for newcomers to pick what to read. Maybe we could sort by title and add sections "Basic" and "Advanced"

nicoddemus replied:
Sounds good.

igortg committed:
Review from @nicoddemus and @marcelotrevisani

- Add a "How to Contribute"
- Separate articles list in sections and sort alphabetically
- Add the talk "Transforming Code into Beautiful, Idiomatic Python"

igortg committed:
Minor style changes
